src/control/cmd/dmg/pretty/network.go

```diff
@@ -40,27 +46,56 @@ func PrintHostFabricMap(hfm control.HostFabricMap, out io.Writer, onlyProviders
 	}
 
 	ew := txtfmt.NewErrWriter(out)
+
 	for _, key := range hfm.Keys() {
+		var hfiMap map[uint32]map[string][]string
+
 		hfs := hfm[key]
 		hosts := control.GetPrintHosts(hfs.HostSet.RangedString(), opts...)
 		lineBreak := strings.Repeat("-", len(hosts))
+		iw := txtfmt.NewIndentWriter(ew, txtfmt.WithPadCount(4))
 		fmt.Fprintf(ew, "%s\n%s\n%s\n", lineBreak, hosts, lineBreak)
-
-		iw := txtfmt.NewIndentWriter(ew, txtfmt.WithPadCount(8))
-		fmt.Fprintf(iw, "Providers: %s\n", strings.Join(hfs.HostFabric.Providers, ","))
 		fmt.Fprintln(ew)
+		fmt.Fprintf(iw, "Available providers: %s\n", strings.Join(hfs.HostFabric.Providers, ", "))
+		fmt.Fprintln(ew)
+
 		if onlyProviders {
 			continue
 		}
 
+		hfiMap = make(map[uint32]map[string][]string)
 		for _, fi := range hfs.HostFabric.Interfaces {
-			fmt.Fprintf(iw, "provider: %s\n", fi.Provider)
-			fmt.Fprintf(iw, "fabric_iface: %s\n", fi.Device)
-			fmt.Fprintf(iw, "pinned_numa_node: %d\n", fi.NumaNode)
-			fmt.Fprintln(ew)
+			if _, ok := hfiMap[fi.NumaNode]; !ok {
+				hfiMap[fi.NumaNode] = make(map[string][]string)
+			}
+			hfiMap[fi.NumaNode][fi.Provider] = append(hfiMap[fi.NumaNode][fi.Provider], fi.Device)
```

Comment:
This is fine... It's not wrong to do all of the map management here, but it's a little distracting from the real work being done in this loop. As a suggestion, you could create a `hfiMap` type alias for `map[uint32]map[string][]string` and put some methods on it to keep the code here cleaner, e.g.:

```go
hfim := make(hfiMap)
for _, fi := range hfs.HostFabric.Interfaces {
        hfim.addInterface(fi)
}
```

The `addInterface()` method could take care of all of the map management stuff and would be easier to unit test as well.

Reply:
Done.